### DISCUSSION TITLE
Category Filter not working properly.

### DISCUSSION THREAD

**Discussion by tayyab2020** [bagisto/bagisto#5352] [General]
**Description**
Category filters are not fetching products if multiple filters are used. For example if I filter product based on only one attribute (Size or Color) than it works fine but when I filter products based on both attributes than it does not fetch any product. According to screenshot attached bellow, product 'XYZ' should have been filtered out in case of multiple attributes filtering.

**Bagisto Version**
1.3.3

![Screenshot 2021-11-21 123224](https://user-images.githubusercontent.com/61475588/142880808-a1778c15-348f-42a0-b569-0e278edb8aae.png)
![Screenshot 2021-11-21 123327](https://user-images.githubusercontent.com/61475588/142880816-2a139bef-2fdf-4513-9d6b-7c8ec8def7ed.png)
![Screenshot 2021-11-21 123401](https://user-images.githubusercontent.com/61475588/142880818-ae4bd729-164a-45ff-8c49-1a4d0b596e55.png)

**Comment by devansh-webkul** [comment]
Hi, @tayyab2020,

I am not able to reproduce this.

**Comment by tayyab2020** [comment]
https://drive.google.com/file/d/1ceISqd4KERzj-44u36Y3cwIhIfxs2umX/view?usp=sharing
I have attached a video, in this video you can see that a product has both color 'Red' and size '240 x 340 cm' attributes but still when i filter it on both attributes (color & size) than i get nothing in the results.

**Comment by tayyab2020** [comment]
I figured out why its not showing results. I have to edit each variant of product and has to select same color and category as of its parent product. Can you do it like this that when a configurable product is created than all of its attributes values (Color, Brand or any custom attribute) and category selected should be same for all its variants by default?

**Comment by tayyab2020** [comment]
Why each variant of a configurable product has not exact same details as that of its parent product?
Like Category of each variant of a product should be stored as that of its main product by default

**Comment by devansh-webkul** [comment]
Each variant value can't be the same because it needs to fill up at the time of configurable product creations. Because variant's actual meaning is different combo whether it is color or size or different things.

If you want to do it, you can customize this at your end.

**Comment by tayyab2020** [comment]
If I create a size variant of a product and than later on I filter category of that product including the size filter and other attributes filters than I don't get that product in results until I edit that product variant and select same attributes as that of its parent product. Kindly let me know in which file I have to do changes so that a variant of the product should have same values as that of its parent product by default. For example if I create a product with variants than all its variants products should have same attributes values as that of its parent product.

**Comment by tayyab2020** [comment]
Lets say I create a product with color attribute value 'White' (product is only in one color so its variants should also be in same color by default) and than I create its size variants, So I have to edit its each variant manually and select color white for it and save it so that it appears in filter page if I use size variant filter along with color filter.

**Comment by devansh-webkul** [comment]
You need to check here for all variant case,
https://github.com/bagisto/bagisto/blob/255f757521b79d8a7d6672f958b5d6aa022b6849/packages/Webkul/Product/src/Type/Configurable.php#L207

**Comment by tayyab2020** [comment]
Thanks for your reply, I actually need to store same information for all variants as of its parent product, where to do that?

**Comment by devansh-webkul** [comment]
Check here, looping of all variants done here,
https://github.com/bagisto/bagisto/blob/255f757521b79d8a7d6672f958b5d6aa022b6849/packages/Webkul/Product/src/Type/Configurable.php#L167

**Comment by tayyab2020** [comment]
Thanks for your help, Kindly let me know do I just have to pass all attribute values (including custom) to "$data['variants']" so all variants of a product have the same attribute values as of its parent product. Or do I have to pass each attribute value manually one by one?

**Reply by devansh-webkul** [reply]
That's all depends on your taste just do it what best you can see for your project.
